### DISCUSSION TITLE
Recommendations on destinations for importing forum SQLite?

### DISCUSSION THREAD

**Discussion by Linnuial** [allquixotic/encuum#14] [Q&A]
Thank you so much for the tool. I was able to scrape all of our forums and store them in the SQLite database.

While I do have some early career history with SQL, I'm wondering if there are any recommended or suggested destination sites which can take the SQLite database as import source?

I'm assuming the author of encuum had some destination in mind... :)

Thanks again!

**Comment by allquixotic** [comment]
I'm not aware of any databases or software that can **directly** ingest the data from encuum unmodified. The API structure of Enjin that produced this data was somewhat unique. Of course, being a proprietary software project, we have no way of knowing if Enjin used some other off-the-shelf product under the hood. We also don't know whether the data as represented in the JSON-RPC API had any correspondence at all to how the data looked in the database on the backend.

I can say that the developer of GuildTag, an Enjin competitor website, is making an effort to support migrating data from users who are currently running Enjin. However, since I do not own a GuildTag website, and I do not work for GuildTag, I cannot attest to the accuracy of the import process, or if there are any "gotchas". You would need to follow the steps to create a GuildTag website, subscribe to the paid service, and import your data. https://guildtag.com/forum-thread/release-notes-v20230304/167055/ https://guildtag.com/forum-thread/release-notes-v20230221/138005/

All this is to say, I don't have any specific recommendations here. If you are intending to resurrect a functional forum with some other forum software, you have a few options:

1. You can create a view that rearranges and renames the columns to match your forum software, then dump that into a bunch of SQL `insert` statements. I have no idea how the unique Enjin IDs pertaining to each row will be compatible with your forum software; this depends on what the forum software is and how it keeps track of its data. For instance, your database might expect UUIDs for forum posts instead of an increasing number.

2. You can write a script that will pull the data out of your sqlite database and somehow get it into the forum where you're deploying. Maybe the forum has an API that you can call to tell it to create posts? Or maybe you have to upload the data into its database (which could be anything).

This is out of the scope of Encuum, because I figure there will be as many paths or "destinations" for the data extracted by Encuum as there are people using it. Each person is going to make a decision on their own whether to host their own server with a forum; use some other guild website service; dump it into a Google Doc; etc. etc. etc. The possibilities are limitless and depending on individual preference.

I'm not going to write a how-to guide on how to host a forum in this issue, because that sort of information is widely available and googleable. You can rent a server in a datacenter; co-locate your own hardware; host from home if you have a stable connection; run something in the cloud (AWS/Azure/Google Cloud, ...); you can host a content management system... like I said, there are infinite possibilities.

If I were to wager a pure guess, something like phpBB or vBulletin is probably a similar enough forum software that a database-to-database import would be ... reasonably easy, _I think_. But don't take that as gospel that you'll have an easy time with either of those pieces of software.

As for my own intentions with how to use the data pulled out of Encuum... I'm not really at liberty to disclose, sorry.

**Comment by Linnuial** [comment]
Thanks so much for the response and the tips!

I mainly was just wondering what your plan was, since you wrote the tool originally for your own purposes. But if you cannot discuss, I understand!

Once again, thank you so much for this tool and for the extended support.  More than anything, I’m just glad to export the data, even if just for archival purposes.

Best of luck to you!
